Sentinel/Classes/DatabaseImportExport/Internal/DatabaseFileManager.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  DatabaseFileManager.swift
+//  Sentinel
+//
+//  Created by Zvonimir Medak on 12.03.2025..
+//
+
+import Foundation
+
+public enum DatabaseFileError: Error {
+    case importFailed
+    case exportFailed
+}
+
+final class DatabaseFileManager {
+
+    // MARK: - Private properties
+
+    private let fileManager: FileManager
+    private let databaseFilePath: String
+
+    private lazy var documentDirectoryUrl: URL? = {
+        fileManager.urls(for: .documentDirectory, in: .userDomainMask).first
+    }()
+
+    private lazy var exportDirectoryUrl: URL? = {
+        documentDirectoryUrl?.appendingPathComponent("export")
+    }()
+
+    private lazy var exportArchiveUrl: URL? = {
+        exportDirectoryUrl?.appendingPathComponent("export.zip")
+    }()
+
+    private lazy var databaseURL: URL? = {
+        guard let documentDirectoryUrl else { return nil }
+        return URL(fileURLWithPath: "\(documentDirectoryUrl.path)/\(databaseFilePath)")
+    }()
+
+    // MARK: - Init
+
+    init(fileManager: FileManager = .default, databaseFilePath: String) {
+        self.fileManager = fileManager
+        self.databaseFilePath = databaseFilePath
+    }
+
+    func importDatabase(from url: URL) throws {
```

Comment:
Based on import/export functionality, do we expect the concrete file here or the `.zip` as well? I would like to handle both cases, because, if we are exporting zip, we could expect to get zip as an import as well.

Reply:
I have added the export as zip, and file but the unzip is complicated. Atleast, from what I have found online, there's no easy was to unzip a file using built-in functions of Swift. Only the archiving is kinda easy with the NSFileCoordinator since the forUploading option zips it.
And I wouldn't want to add a third party too just so we can accept a zip since I believe mostly people will just export the file or zip (depending on the size), and import the file.

Do you have an example or an idea how we can achieve the unarchiving using built-in functions?

Reply:
I agree with you 👍